KeyboardAvoidanceSwiftUI/KeyboardAdaptive.swift

```diff
@@ -12,24 +12,38 @@ import Combine
 /// Note that the `KeyboardAdaptive` modifier wraps your view in a `GeometryReader`, 
 /// which attempts to fill all the available space, potentially increasing content view size.
 struct KeyboardAdaptive: ViewModifier {
+    private let rerender: AnyPublisher<Any, Never>
     @State private var bottomPadding: CGFloat = 0
-
+    
+    init(rerender: AnyPublisher<Any, Never>? = nil) {
+        var publishers = [Publishers.keyboardHeight.map { arg -> Any in arg }.eraseToAnyPublisher()]
+        if let rerender = rerender {
+            publishers.append(rerender)
+        }
+        self.rerender = Publishers.MergeMany(publishers).eraseToAnyPublisher()
+    }
+    
+    private func bottomPadding(forGeometry geometry: GeometryProxy) -> CGFloat {
+        let keyboardTop = geometry.frame(in: .global).height - Publishers.keyboardHeight.value
+        var focusedTextInputBottom = UIResponder.currentFirstResponder?.globalFrame?.maxY ?? 0
```

Comment:
It would be really nice if we could just observe the frame of this translating, that's the real condition that should prompt us to re-pad, without the client having to do anything. Maybe something using KVO?